src/query.rs

```diff
@@ -0,0 +1,77 @@
+//! Constants corresponding to standard HTTP names.
```

Comment:
I found the level of nesting pretty inconvenient to navigate through in the API documentation, and several modules have only one item. One or two modules should be sufficient, with constants named as e.g. `CONTENT_TYPE_JSON`, `PARAM_CLIENT_ID`, `PARAM_SCOPE` depending on how they are broken up.

Reply:
I considered that extra-nesting for values would make the structure more organized and resolve ambiguities as it is not obvious which part of the constant name corresponds to the header and which one -- to the value thus (in theory) there could be a problem if some param/header had the name like `Foo-bar` with common value `baz` and another one -- name `Foo` and value `bar/baz` in this case both constants should have to be named in `FOO_BAR_BAZ`.

I've also discovered that I've accidentally used the incorrect name for `APPLICATION_JSON` constant as I've includede header name prefix (`ACCEPT_`) in it which was redundant as it already is in `accept` module. Fixed in 37e2a0c

Thus, I would suggest sticking to namespace-per-header/parameter but if you insist then I will add a commit flattening the structure to only `header` and `parameter` modules.

Reply:
> Thus, I would suggest sticking to namespace-per-header/parameter but if you insist then I will add a commit flattening the structure to only header and parameter modules.

Please do at least flatten the `header` module; `param` seems fine. Each additional module layer is an additional page in the documentation and an additional click to discover the full list of constants that are available, and there isn't enough content to justify the additional layers here. I'm just as concerned about needlessly frustrating users who need constants that *aren't* in this list, by making them click through 4 screens to find out that they are missing.

Ideally, all of the available constants including their values should fit onto one or two pages of documentation.

Reply:
> Please do at least flatten the header module; param seems fine.

Done in 6649919